### ISSUE TITLE
Support liniting inside inline macros arguments

### ISSUE THREAD

**Issue by MKowalski8** [software-mansion/cairo-lint#333]
### Problem

Due to recent compiler changes related to inline macros, the linter currently cannot analyze code inside macro arguments. As a result, any diagnostics within inline macros like `assert!` are not reported.

For example:
``` cairo
assert!(if x > y || x < y {
    true
} else {
    false
});

assert!({((a)) == b});

assert!(if true == true {
    loop {
        let a: felt252 = 'hello';
        let b = a.clone();
        println!("{}", b);
        break ();
    }
    
    true
} else {
    true
});

assert!({
    let a: felt252 = 'hello';
    let a: Result<[u64; 2], felt252> = Result::Ok([10, 20]);
    let _x = match a {
        Result::Ok(v) => {
            v
        },
        Result::Err(_) => 
            [1, 2]
    };
    let b = a.clone(); 
    true
});

println!("{}", x / x);
```

Even though many issues are present, the linter is currently unable to report them.

### Notes

One notable change is that arguments in `Syntax` are now only represented as `tokens`.  As a result, accessing this data directly at the `Syntax` level is likely no longer possible.

```cairo
assert!((((a))) == b);
```

<img width="673" alt="Image" src="https://github.com/user-attachments/assets/14c46f73-033d-4374-9ed7-d8a946c09940" />

**Comment by wawel37** [comment]
As we've researched, there is possibility to extract arguments from inline macros, but only from the legacy ones. It's also not a good idea to lint args in all the inline macros, as there might be some arbitrary code, which would not be valid by itself. 

There is a also a major problem, because the `as_legacy_inline_macro` nodes have different parents and textoffsets than it's original nodes.
